src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -1410,6 +1410,26 @@ void TaskSketcherElements::onSelectionChanged(const Gui::SelectionChanges& msg)
                         }
                     }
                 }
+                else if (shapetype.size() > 12 && shapetype.substr(0, 12) == "ExternalEdge") {
+                    QRegularExpression rx(QString::fromLatin1("^ExternalEdge(\\d+)$"));
+                    QRegularExpressionMatch match;
+                    boost::ignore_unused(expr.indexOf(rx, 0, &match));
+                    if (match.hasMatch()) {
+                        bool ok;
+                        int ElementId = -match.captured(1).toInt(&ok) - 2;

```

Comment:
The leading minus sign looks like a mistake at first glance -- can you add a comment to explain?

Reply:
Hi @chennes!

Thanks for looking at this.

AFAICS the minus sign is needed in case of ExternalEdges:
`ElementId` is matched against `item->ElementNbr` a few lines after in this PR (line 1425).

The value of `item->ElementNbr` is set when a `ElementItem* item` is initialized by the method `TaskSketcherElements::slotElementsChanged`
https://github.com/FreeCAD/FreeCAD/blob/62463fa0bd2c6962550ef67f94ca527b81b21cd9/src/Mod/Sketcher/Gui/TaskSketcherElements.cpp#L1889
and as you can see that value is negative and starts from -3
https://github.com/FreeCAD/FreeCAD/blob/62463fa0bd2c6962550ef67f94ca527b81b21cd9/src/Mod/Sketcher/Gui/TaskSketcherElements.cpp#L1843

If you want I can add a comment but in that case I would say that it should be better to add it also in other parts of the code related to the ExternalElements where the Element Number / Index is valued with the minus sign.

In any case just let me know how you prefer to proceed and I'll apply the modifications needed.

Thanks again!